core/src/main/java/com/linecorp/armeria/internal/server/JacksonUtil.java

```diff
@@ -16,40 +16,32 @@
 
 package com.linecorp.armeria.internal.server;
 
-import java.lang.reflect.InvocationTargetException;
+import static java.util.Objects.requireNonNull;
+
 import java.util.List;
+import java.util.ServiceLoader;
 
 import com.fasterxml.jackson.databind.Module;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.json.JsonMapper;
-import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
 
-public final class JacksonUtil {
+import com.linecorp.armeria.server.JacksonModuleProvider;
 
-    private static final List<Module> defaultModules;
-
-    static {
-        final List<String> additionalModules = ImmutableList.of(
-                "com.fasterxml.jackson.module.scala.DefaultScalaModule",
-                "com.fasterxml.jackson.module.kotlin.KotlinModule");
-
-        // Add the additional modules if they are in the classpath
-        final ImmutableList.Builder<Module> moduleBuilder = ImmutableList.builderWithExpectedSize(2);
-        for (String moduleClassName : additionalModules) {
-            try {
-                final Class<?> moduleClass = Class.forName(moduleClassName);
-                final Module module = (Module) moduleClass.getDeclaredConstructor().newInstance();
-                moduleBuilder.add(module);
-            } catch (ClassNotFoundException | NoSuchMethodException | InstantiationException |
-                    IllegalAccessException | InvocationTargetException ignored) {
-            }
-        }
-        defaultModules = moduleBuilder.build();
-    }
+public final class JacksonUtil {
 
     public static ObjectMapper newDefaultObjectMapper() {
         final JsonMapper.Builder jsonMapperBuilder = JsonMapper.builder();
-        defaultModules.forEach(jsonMapperBuilder::addModule);
+        final ServiceLoader<JacksonModuleProvider> providers = ServiceLoader.load(JacksonModuleProvider.class);
+        if (Iterables.isEmpty(providers)) {
+            jsonMapperBuilder.findAndAddModules();
+        } else {
+            for (JacksonModuleProvider provider : providers) {
```

Comment:
Let's leave a log as we do in https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/SerializationFormat.java#L81
so that a user can notice the module is applied.

Reply:
Nice suggestion.